### ISSUE TITLE
Impossible to create AWS::Serverless::Api Auth property conditionally

### ISSUE THREAD

**Issue by akornato** [aws/serverless-application-model#1955]
### Description:
Impossible to assign `!Ref AWS::NoValue` to `AWS::Serverless::Api` `Auth` property.

### Steps to reproduce:
```yaml
ResourceName:
  Type: AWS::Serverless::Api
  Properties:
    StageName: Prod
    Auth: !Ref AWS::NoValue
```
or
```yaml
ResourceName:
  Type: AWS::Serverless::Api
  Properties:
    StageName: Prod
    Auth: !If [someCondition, !Ref AWS::NoValue, !Ref AWS::NoValue]
```

### Observed result:
AWS SAM fails to create changeset with `Invalid value for 'Auth' property` error.

### Expected result:
`!Ref AWS::NoValu` should remove `Auth` property altogether (at least if used with `!If`) according to:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html

This is required to create `Auth` property conditionally, e.g. in non-Prod env only.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. MacOS
2. SAM CLI version 1.15.0
3. AWS region eu-west-1

**Comment by wchengru** [comment]
Thanks for reporting this!
I have reproduced this issue. Similar to another issue (#1435 ) in `Domain`, currently AWS::NoValue is not supported in `Auth`. I will also labeling this one for internal review.

**Comment by barnou** [comment]
Hello!

We are also blocked by this outcome.
We would like to make the `ResourcePolicy` property conditional, but this is currently not possible. 

Do you have any information on how to resolve this issue?

**Comment by xazhao** [comment]
Thanks for asking. I already added it to our backlog for internal review.

**Comment by lucascott** [comment]
Hello!

We're blocked as well.
The same issue emerges with the `Role` property of `AWS::Serverless::Function` resource.

**Comment by AzherAleem** [comment]
Hi, Is there is any update or workaround for this? I have been trying to set the Auth for API conditionally but it does not work:
```

ConnectAPI:
    Type: AWS::Serverless::Api
    Properties:
      EndpointConfiguration:
        Type: REGIONAL
      Name: !Join
        - '-'
        - - !Ref NamePrefix
          - 'ConnectAPI'
          - !Ref Environment
      StageName: !Ref Environment
      Auth:
        !If
          - SPOTCognitoUserPoolARNPassedInParameters
          - 
            DefaultAuthorizer: MyCognitoAuthorizer
            Authorizers:
              MyCognitoAuthorizer:
              UserPoolArn: !Ref SPOTCognitoUserPoolARN
          - !Ref "AWS::NoValue"  

```

Above is my implementation which does not work in case the parameter is not provided.

**Comment by tharaka-wanigasekera-ofx** [comment]
Hello!

Are there any updates?

We have a similar issue when using `!FindInMap`

```
ResourcePolicy:
   IpRangeWhitelist: !FindInMap
      - EnvironmentMappings
      - !Ref Environment
      - ipRangeWhitelist
```

**Comment by hoffa** [comment]
Closing in favor of https://github.com/aws/serverless-application-model/issues/2533.
